### DISCUSSION TITLE
asdf interaction - "unknown command: node. Perhaps you have to reshim?"

### DISCUSSION THREAD

**Discussion by yeraydiazdiaz** [tox-dev/tox#2247] [Q&A] [answered]
Hey all, in lunr.py I have some acceptance tests that compare the results between the nodejs and the Python versions.

I switched recently from pyenv + nvm to asdf and it works fine for the most part, except when trying to run these acceptance tests inside the tox env. It seems `node` is not shimmed correctly inside the tox env for some reason. I can see tox finds the `node` executable in $PATH it fails with "unknown command: node. Perhaps you have to reshim?".

I've tried adding `node` to the allow_externals and using `passenv` to include any env vars that seem related to asdf but I can't get it to work.

Any ideas?

**Comment by gaborbernat** [comment]
> It seems `node` is not shimmed correctly inside the tox env for some reason

tox isolates the tools from your environment variables, so that's likely the problem. I don't know what environment variables asdf relies on but you'll have to identify that and pass it to the ``passenv`` of the environment. You'd need to provide more details to find out what's the problem, for example, a docker image replicating the issue or a detailed log messages.

**Reply by yeraydiazdiaz** [reply]
Thanks @gaborbernat, the only env var that seemed relevant was `ASDF_DIR` but that didn't help. But I managed to work around it by using https://github.com/asdf-community/asdf-direnv which 'fixes' the paths and bypasses the asdf shims.

**Comment by yeraydiazdiaz** [comment, marked as answer]
Fixed using https://github.com/asdf-community/asdf-direnv as described above.

**Comment by groteck** [comment]
Not working for `poetry` installed with `asdf-poetry` plugin

**Reply by looztra** [reply]
made it work with:

``` 
setenv =
    ASDF_DATA_DIR = {env:ASDF_DIR}
```

**Reply by looztra** [reply]
Actually, it works better with

```
setenv =
    ASDF_DATA_DIR = {env:HOME}/.asdf
```

because it is compatible with git and brew installs (it must point towards the directory that contains the plugins)
